packages/core/saved-objects/core-saved-objects-base-server-internal/src/model_version/version_compare.test.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { compareModelVersions } from './version_compare';
+
+describe('compareModelVersions', () => {
+  it('returns the correct value for greater app version', () => {
+    const result = compareModelVersions({
+      appVersions: {
+        foo: 3,
+        bar: 2,
+      },
+      indexVersions: {
+        foo: 2,
+        bar: 2,
+      },
+      deletedTypes: [],
+    });
+
+    expect(result.status).toEqual('greater');
+  });
+
+  it('returns the correct value for lesser app version', () => {
+    const result = compareModelVersions({
+      appVersions: {
+        foo: 1,
+        bar: 2,
+      },
+      indexVersions: {
+        foo: 2,
+        bar: 2,
+      },
+      deletedTypes: [],
+    });
+
+    expect(result.status).toEqual('lesser');
+  });
+
+  it('returns the correct value for equal versions', () => {
+    const result = compareModelVersions({
+      appVersions: {
+        foo: 2,
+        bar: 2,
+      },
+      indexVersions: {
+        foo: 2,
+        bar: 2,
+      },
+      deletedTypes: [],
+    });
+
+    expect(result.status).toEqual('equal');
+  });
+
+  it('handles new types not being present in the index', () => {
+    const result = compareModelVersions({
+      appVersions: {
+        foo: 2,
+        new: 1,
+      },
+      indexVersions: {
+        foo: 2,
+      },
+      deletedTypes: [],
+    });
+
+    expect(result.status).toEqual('greater');
+  });
+
+  it('handles types not being present in the app', () => {
+    const result = compareModelVersions({
+      appVersions: {
+        foo: 3,
+      },
+      indexVersions: {
+        foo: 2,
+        old: 1,
+      },
+      deletedTypes: [],
+    });
+
+    expect(result.status).toEqual('conflict');
```

Comment:
I'm probably missing something here, but why is this test behaving differently from the last one?